File: 프로그래머스/lv1/김원래/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    
    
    for(vector<int>& command : commands) {
        vector<int> _substr;
        int i = command[0];
        int j = command[1];
        int k = command[2];
        
        for(int idx = i -1 ; idx < j ; ++idx) {
            _substr.push_back(array[idx]);
        }
        sort(_substr.begin(), _substr.end());
        
        answer.push_back(_substr[k-1]);
    }
    
    return answer;
}

/*



*/

File: 프로그래머스/lv1/이재표/K번째수.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int idx=0;
        for(int i=0;i<commands.length;i++){
            int[]split=Arrays.copyOfRange(array,commands[i][0]-1,commands[i][1]);
            Arrays.sort(split);
            answer[idx++]=split[commands[i][2]-1];
        }
        
        return answer;
    }
}
